### PR TITLE
security assessment proofread and benefits editing

### PR DESCRIPTION
**Pull request by itaysk**
While working on https://github.com/cncf/sig-security/issues/448 and https://github.com/cncf/sig-security/issues/445 I have read the relevant readmes, and made a couple of commits:
First there's a straightforward proofread, fixing typos, simplification, etc.
Then there's the content change around the goals and benefits.
There is an overlap in content that describes the motivation and goals of the assessment, between the main assessments/README.md file (in the Goals section) and the guide/README.md (in the opening sentence and the closing "benefits of a security assessment" section. I think the "guide" should be a technical document that describes the process. 
I have pulled the relevant content from the guide to the readme and added insights about the benefits to the projects, and relation to the TOC process.

### CODE REVIEW

**Comment by itaysk** [comment]
besides the fixed issues, there's the note at the end with didn't sit well for me:

> we expect quick turnaround (at most a week)

If the meaning is for the entire assessment then this conflicts with previously stated:

> *3 week* timeframe for assessments

and if it relates to the "security review (5)" phase then we should reconcile with the stated

> at least one week for review

File: assessments/README.md [hunk 1]
```diff
@@ -2,45 +2,50 @@
...
-     profile with the project’s risk profile and effectively allocate resources
-     (for review and needed project contribution)
-   * Structured metadata allows for navigation, grouping and cross-linking
+   * Clear rubric for security profile enables potential users to align their risk
+     profile with the project’s risk profile
 
-We expect that this process will raise awareness of how specific open source
-projects affect the security of a cloud native system; however, separate
-activities may be needed to achieve that purpose using materials generated by
-the assessements.
```
Comment (itaysk) [inline assessments/README.md:39]: This paragraph felt devoid for me, but please let me know if I was missing something important there

**Comment by TheFoxAtWork** [comment]
@itaysk i think some of these are resolved with the proposed updates and new docs from #479 & #488   would you review these, check that the suggestions/intent of your updates are included or missed?  We'd like to update the entire process with the new suggestions from the security assessment working group and these PRs need another set of eyes.  @lumjjb thoughts?

**Comment by itaysk** [comment]
I was not aware such extensive restructure is undergoing, closing this until those PR are merged and then I'll take another look

**Pull request closed without merging**